Mathlib/Combinatorics/Enumerative/Bell.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir & María-Inés de Frutos—Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María-Inés de Frutos—Fernández
+-/
+
+import Mathlib.Data.Nat.Choose.Multinomial
+
+/-! # Bell numbers for multisets
+
+For `n : ℕ`, the `n`th Bell number is the number of partitions of a set of cardinality `n`.
+Here, we define a refinement of these numbers, that count, for any `m : Multiset ℕ`,
+the number of partitions of a set of cardinality `m.sum` whose parts have cardinalities
+given by `m`.
+
+The definition presents it as an integer.
+
+* `Multiset.bell`: number of partitions of a set whose parts have cardinalities a given multiset
+
+* `Multiset.bell_zero`: `Multiset.bell 0 = 1`
```

Comment:
I'm not super convinced very basic lemmas like `bell_zero` need mentioning in the module doc. They can already efficiently be found through the symbol-reading convention.